### PR TITLE
Fixes #10889 - Allow selection of resource pool

### PR DESCRIPTION
**Pull request by shlomizadok**

### CODE REVIEW

**Comment by ohadlevy** [comment]
@shlomizadok note failing tests.

**Comment by shlomizadok** [comment]
Added permissions, thanks for the headsup

File: config/routes.rb [hunk 1]
```diff
@@ -331,6 +331,7 @@
         member do
           post 'template_selected'
           post 'cluster_selected'
+          post 'resource_pools'
```
Comment (dLobatog) [inline config/routes.rb]: I realize the other AJAX actions are `post` too, but for an action that is purely retrieving data (the resource pools), I don't see why not using GET with params?

File: app/controllers/compute_resources_controller.rb [hunk 2]
```diff
@@ -112,13 +112,21 @@ def cluster_selected
     end
   end
 
+  def resource_pools
```
Comment (dLobatog) [inline app/controllers/compute_resources_controller.rb:114]: Isn't this method equivalent to `api/v2/compute_resources/:id/available_clusters/:cluster_id/available_resource_pools"`?

**Comment by dLobatog** [comment]
@shlomizadok Thanks, I've left a couple of comments. I'll test after we clarify whether to use the existing API endpoint (available_resource_pools) or if there's any benefit for adding an action to the regular controller.

File: app/controllers/compute_resources_controller.rb (hunk 2 above)
Comment (shlomizadok) [inline app/controllers/compute_resources_controller.rb:114]: API is not working with nested clusters (how do I send the cluster: "NY/Manhattan/Marketing" via api?), which is why it is better to send the cluster as a parameter (not uri param)

**Comment by shlomizadok** [comment]
@eLobato thanks for reviewing.
I am not using the API as it is not usable for nested clusters.
I have changed the ajax call from `post` to `get`.
I have added support from resource pool when creating a vm from a template (e.g., clone)

**Comment by shlomizadok** [comment]
re[test] - random katello failure

**Comment by domcleal** [comment]
If you're adding a new separately precompiled JS file for compute resources, note that it's used on many more pages than the couple you've added it to (search for `javascript.*compute_resource app/views` to see), it's probably better just to merge/require it in compute_resource.

**Comment by shlomizadok** [comment]
merged into compute_resource.

**Comment by shlomizadok** [comment]
Anything missing from this PR? can you review and test please?

**Comment by timogoebel** [comment]
I tested this PR. Works quite well. :+1:

**Comment by dLobatog** [comment]
@shlomizadok Ready to merge in my opinion, it just lacks a test for the `resource_pools` action in the controller. You can probably use `Fog.mock!` to test that it works well.

**Comment by shlomizadok** [comment]
@dLobatog - thanks for reviewing. Added some tests, hope this is what you had in mind.
(Please note that Fog.mock for resource_pools returns nothing, so I stubbed it.)

**Comment by dLobatog** [comment]
Merged as 55e5f207437f30fc7ff0ed22fc90bace21dee574, thanks @shlomizadok!

**Pull request closed without merging**